frontend/src/components/App.tsx

```diff
@@ -1,21 +1,70 @@
+import * as PageConstants from "./pages/page_constants";
 import * as React from "react";
+import AppBar from "./appbar/AppBar";
+import { BrowserRouter } from "react-router-dom";
+import clsx from "clsx";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import Drawer from "./drawer/Drawer";
+import { Explore } from "./pages/explore/Explore";
 import { hot } from "react-hot-loader";
+import { Login } from "./pages/login/Login"; 
+import { Profile } from "./pages/profile/Profile";
+import { Route } from "react-router-dom";
+import { Switch } from "react-router-dom";
+import { YourClubs } from "./pages/your_clubs/YourClubs";
 
-const reactLogo = require("./../assets/img/react_logo.svg");
-import "./../assets/scss/App.scss";
+interface AppProps {}
 
-class App extends React.Component<{}, undefined> {
-    public render() {
-        return (
-            <div className="app">
-                <h1>Hello World!</h1>
-                <p>Foo to the barz</p>
-                <img src={reactLogo.default} height="480"/>
-            </div>
-        );
-    }
+interface AppState {
+  /** @state determines if drawer is shown or not*/ open: boolean;
 }
 
-declare let module: object;
+class App extends React.Component<AppProps, AppState> {
+
+  constructor(props) {
+    super(props);
+    
+    this.state = {
+      open: false
+    };
+  }
+
+  // Handles opening drawer in AppBar and Drawer components
+  handleDrawerOpen = () => {
+    this.setState({open: true});
+  };
 
+  // Handles closing drawer in AppBar and Drawer components
+  handleDrawerClose = () => {
+    this.setState({open: false});
+  };
+  
+  public render() {
+    return (
+      <BrowserRouter>
+        <CssBaseline />
+        <AppBar 
+          open={this.state.open}
+          handleDrawerOpen={this.handleDrawerOpen.bind(this)}
```

Comment:
nit: You shouldn't need the `.bind(this)` part because arrow functions are already bound to `this`

Reply:
I didn't know that; that clarifies a lot of what I've been reading. Thanks!